imageregistry/v1/stable.config.testsuite.yaml

```diff
@@ -16,3 +16,72 @@ tests:
         logLevel: Normal
         operatorLogLevel: Normal
         replicas: 1
+  - name: Should be able to configure internal storage network access for Azure
+    initial: |
+      apiVersion: imageregistry.operator.openshift.io/v1
+      kind: Config
+      spec:
+        replicas: 1
+    expected: |
+      apiVersion: imageregistry.operator.openshift.io/v1
+      kind: Config
+      spec:
+        logLevel: Normal
+        operatorLogLevel: Normal
+        replicas: 1
+        storage:
+          azure:
+            type: Internal
+            vnetName: foo-bar
+            subnetName: bar_foo
+            privateEndpointName: my-private-endpoint
+  - name: Should be able to configure internal storage network access for Azure without details
+    initial: |
+      apiVersion: imageregistry.operator.openshift.io/v1
+      kind: Config
+      spec:
+        replicas: 1
+    expected: |
+      apiVersion: imageregistry.operator.openshift.io/v1
+      kind: Config
+      spec:
+        logLevel: Normal
+        operatorLogLevel: Normal
+        replicas: 1
+        storage:
+          azure:
+            type: Internal
+  - name: Should reject configuration with invalid vnet name in internal storage network access config
```

Comment:
"Should reject" are expected to fail. `expectedError` is expected to be set and an error message produced.

Reply:
https://github.com/openshift/api/tree/master/tests and https://github.com/openshift/api/blob/master/tests/Makefile#L9

Reply:
This was weird - giving an invalid input for `subnetName` or `vnetName` did not cause any errors - what happened instead was that the fields were removed (see updated test case if this doesn't make sense). So I removed the `expectedError`.

Reply:
```
networkAccess:
  type: Internal
  internal:
    vnetName: .invalid
```
is the right form

Reply:
aha! thank you.
but so when given a field that doesn't exist, it just fails silently? is that expected?

Reply:
Yeah, fields without mapping get ignored. I think CRDs have a special configuration that allow to fail for missing mapping. Though, a plain json does not have any such constraint.

Reply:
I see, too bad. Thanks for explaining.